scripts/regenerate_test_rustdocs.sh

```diff
@@ -3,34 +3,32 @@
 # Fail on first error, on undefined variables, and on failures in pipelines.
 set -euo pipefail
 
-# Go to the test_crates directory.
-cd "$(git rev-parse --show-toplevel)/test_crates"
-
 export CARGO_TARGET_DIR=/tmp/test_crates
-RUSTDOC_OUTPUT="$CARGO_TARGET_DIR/doc/test_crates.json"
-TARGET_DIR="$(git rev-parse --show-toplevel)/localdata/test_data"
+RUSTDOC_OUTPUT_DIR="$CARGO_TARGET_DIR/doc"
+TOPLEVEL="$(git rev-parse --show-toplevel)"
+TARGET_DIR="$TOPLEVEL/localdata/test_data"
 
 # Allow setting an explicit toolchain, like +nightly or +beta.
 set +u
 TOOLCHAIN="$1"
 set -u
 RUSTDOC_CMD="cargo $TOOLCHAIN rustdoc"
 
-# Ensure the target test data directory exists.
-mkdir -p "$TARGET_DIR"
-
-# Make the baseline configuration file.
-echo "Generating: baseline"
-RUSTC_BOOTSTRAP=1 $RUSTDOC_CMD -- -Zunstable-options --output-format json
-mv "$RUSTDOC_OUTPUT" "$TARGET_DIR/baseline.json"
-
-# For each feature, re-run rustdoc with it enabled.
-features="$(cargo metadata --format-version 1 | \
-    jq --exit-status -r '.packages[] | select(.name = "test_crates") | .features | keys[]')"
-while IFS= read -r feat; do
-    echo "Generating: $feat"
-    RUSTC_BOOTSTRAP=1 $RUSTDOC_CMD --features "$feat" -- -Zunstable-options --output-format json
-    mv "$RUSTDOC_OUTPUT" "$TARGET_DIR/$feat.json"
-done <<< "$features"
+# Run rustdoc on test_crates/*/{new,old}/
+for crate_pair in $(find "$TOPLEVEL/test_crates/" -maxdepth 1 -mindepth 1 -type d); do
+	# Removing path prefix, leaving only the directory name without forward slashes
+	crate_pair=${crate_pair#"$TOPLEVEL/test_crates/"}
```

Comment:
Why shouldn't this be quoted? Maybe the `crate_pair` would have spaces in it, for whatever reasons. Though I hope that the quotation marks match with each other correctly in the old version.

Reply:
I think it should be quoted -- my suggested edit just wraps the whole thing in one extra set of quotes. That's just something I do obsessively in bash scripts since I've debugged several too many scripts where missing quotes either broke something (less bad option) or enabled script injection (much worse option). I just was wondering if adding the extra quotes might break something that I didn't notice.

Reply:
Oh sorry, I've misunderstood. I thought you wanted to delete the quotes. I'll change it in a separate PR.